lib/miq_automation_engine/service_models/miq_ae_service_container_deployment.rb

```diff
@@ -0,0 +1,172 @@
+          unless res.include?("0\r\n")
+            conneted = false
+            unreachable_ips << sub_ip
+          end
+        end
+        [conneted, unreachable_ips]
+      end
+      unless success
+        raise StandardError, "couldn't connect to : #{unreachable_ips.join(',')}"
+      end
+      success
+    end
+
+    def playbook_running?
+      pid = ar_method do
+        @object.customizations[:agent][:agent_pid] if @object.customizations[:agent]
+      end
+      !pid.nil?
+    end
+
+    def run_playbook_command(ip, username, cmd)
+      result = {:finished => false}
+      playbook_pid = ar_method do
+        @object.customizations[:agent][:playbook_pid] if @object.customizations[:agent]
+      end
+      ssh = LinuxAdmin::SSH.new(ip, username, ssh_auth.auth_key)
+      if playbook_pid
+        process_running = ssh.perform_commands(["sudo kill -0 #{playbook_pid}"])[:stdout]
+        unless process_running.empty?
+          result[:finished] = true
+          result[:stdout] = ssh.perform_commands(["sudo cat /tmp/ansible.log"])[:stdout]
+          stop_agent(ssh)
+        end
+      else
+        pid, socket = create_agent(ssh, username)
+        ssh.perform_commands(["sudo rm -f /tmp/*ansible.log*"])
+        ssh.perform_commands(["SSH_AUTH_SOCK=#{socket} nohup sudo -b -E #{cmd}"])
+        playbook_pid = ssh.perform_commands(["sudo pgrep -f 'sudo -b -E'"])[:stdout].split("\r").first
+        customize(:playbook_pid => playbook_pid, :agent_pid => pid, :agent_socket => socket)
+      end
+      result
+    end
+
+    def agent_exists?
+      ar_method do
+        @object.customizations[:agent][:agent_pid] if @object.customizations[:agent]
+      end
+    end
+
+    def create_agent(ssh, username)
+      output = ssh.perform_commands(["ssh-agent -a #{"ssh_manageiq_#{id}"}"])[:stdout]
+      socket = output.split('=')[1].split(' ')[0].chop
+      socket = "/#{username}/#{socket}"
+      socket = "/home#{socket}" if username != "root"
+      pid = output.split('=')[2].split(' ')[0].chop
+      ssh.perform_commands(["SSH_AUTH_SOCK=#{socket} SSH_AGENT_PID=#{pid} ssh-add -"], nil, ssh_auth.auth_key)
+      [pid, socket]
+    end
+
+    def stop_agent(ssh)
+      pid =  ar_method do
+        @object.customizations[:agent][:agent_pid]
+      end
+      socket = ar_method do
+        @object.customizations[:agent][:agent_socket]
+      end
+      ssh.perform_commands(["SSH_AGENT_PID=#{pid} ssh-agent -k &> /dev/null"])
+      ssh.perform_commands(["sudo rm -f #{socket}"])
+      customize(:playbook_pid => nil, :agent_pid => nil, :agent_socket => nil)
+    end
+
+    def analyze_ansible_output(output)
+      if output.empty?
+        return false
+      end
+      results = output.rpartition('PLAY RECAP ********************************************************************').last
+      results = results.split("\r\n")
+      results.shift
+      passed = true
+      results.each do |node_result|
```

Comment:
'passed' can be simplified to
results.detect { |x| !x.include?("unreachable=0") || !node_result.include?("failed=0") }.blank?

right?